### PR TITLE
feat: spinner component creation

### PR DESCRIPTION
**Pull request by MarcosOliveir4**
## Description

This PR adds the ion-spinner component, the ion-spinner can be used to indicate loading or progress in different parts of the application and can be styled to suit the application's design. ion-spinner is easy to use and can be added to any page or component. This feature enhances the user experience and provides visual feedback for long running operations.

## Proposed Changes

- Feature ion-spinner

## Screenshots

Ion-_-Feedback-_-Spinner-Spinner-Primary-⋅-Storybook

## Compliance

- [x] I have verified that this change complies with our code and contribution policies.
- [x] I have verified that this change does not cause regressions and does not affect other parts of the code.

### CODE REVIEW

**Review by AlyssonMascarenhas — APPROVED** [review]
Nice work <3

**Comment by AlyssonMascarenhas** [comment]
Buuuut,
i miss PR description :(

File: projects/ion/src/lib/spinner/spinner.componet.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      expect(screen.getByTestId('ion-spinner')).toHaveClass(
+        `ion-spinner ${color}`
+      );
+    }
+  );
+
+  it('should render SpinnerComponent with custom color', async () => {
+    await sut({ ...defaultProps, customColor: '#c05bff' });
+    expect(screen.getByTestId('ion-spinner')).toHaveStyle({
+      'border-left-color': '#c05bff',
```
Comment (iurynogueira) [inline projects/ion/src/lib/spinner/spinner.componet.spec.ts:33]:
```suggestion
    await sut({ ...defaultProps, customColor});
    expect(screen.getByTestId('ion-spinner')).toHaveStyle({
      'border-left-color': customColor
```

File: projects/ion/src/lib/spinner/spinner.componet.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    });
+  });
+
+  it('should render SpinnerComponent with custom size', async () => {
+    await sut({ ...defaultProps, size: 48 });
+    expect(screen.getByTestId('ion-spinner')).toHaveStyle({
+      width: '48px',
+      height: '48px',
+    });
```
Comment (iurynogueira) [inline projects/ion/src/lib/spinner/spinner.componet.spec.ts:42]: same here

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**